### PR TITLE
Additional remarks 781 form

### PR DESCRIPTION
**Pull request by imeecuison**
## Description
Textarea box for user to input additional remarks.

## Testing done
tests/pages/additionalRemarks781.unit.spec.jsx

## Screenshots
screen shot 2018-11-23 at 1 06 50 pm

## Acceptance criteria
- [ ] User is able to answer “If there is anything else you would like to tell us about the stressful events that contributed to your PTSD, you can provide that information below. If you have any supporting documents or buddy statements to support your claim, you’ll have a chance to upload those later in the application.”
- [ ] User is able to continue to the next step in the process
**Note: Field is not required.**

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by aaronjlech — COMMENTED** [review]
This looks good, just the 1 minor comment on the depends function. Will there be an additional remarks for 781a? If so we can probably make this 1 branch?

File: src/applications/disability-benefits/all-claims/config/form.js [hunk 1]
```diff
@@ -259,6 +264,14 @@ const formConfig = {
           uiSchema: uploadPtsdDocuments.uiSchema,
           schema: uploadPtsdDocuments.schema,
         },
+        additionalRemarks781: {
+          title: 'Additional Remarks - 781',
+          path: 'new-disabilities/additional-remarks-781',
+          depends: formData =>
+            needsToEnter781(formData) && isAnsweringPtsdForm(formData),
```
[the remark this answers is not in the document]
Comment (imeecuison) [reply src/applications/disability-benefits/all-claims/config/form.js]: Removed 'needsToEnter781'. For 781a: the additional remarks equivalent has already been done: additional remarks - physical, emotional, work, etc. Thanks!

**Comment by imeecuison** [comment]
@aaronjlech : Can you check this one again? Thank you!

File: src/applications/disability-benefits/all-claims/config/781/index.js [hunk 2]
```diff
@@ -100,13 +93,6 @@ export function createFormConfig781a(iterations) {
         uiSchema: incidentSupport.uiSchema('781a'),
         schema: incidentSupport.schema,
       },
-      [`secondaryIncidentDate${index}`]: {
-        title: `${numberToWords[index]} PTSD assault incident date`,
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/config/781/index.js]: Can we put this title back on the one that is in the file?

File: src/applications/disability-benefits/all-claims/config/781/index.js [hunk 3]
```diff
@@ -52,13 +52,6 @@ export function createFormConfig781(iterations) {
         uiSchema: incidentSupport.uiSchema('781'),
         schema: incidentSupport.schema,
       },
-      [`incidentDate${index}`]: {
-        title: `${numberToWords[index]} PTSD incident date`,
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/config/781/index.js]: Same with this one, put this title on the incidentDate page that is in the file.

File: src/applications/disability-benefits/all-claims/config/form.js [hunk 4]
```diff
@@ -531,4 +540,4 @@ const formConfig = {
   },
 };
 
-export default formConfig;
+export default createFormConfig;
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/config/form.js]: change this back to `formConfig`.

File: src/applications/disability-benefits/all-claims/config/form.js [hunk 5]
```diff
@@ -93,7 +95,7 @@ import { PTSD, PTSD_INCIDENT_ITERATION } from '../constants';
 
 import fullSchema from './schema';
 
-const formConfig = {
+const createFormConfig = {
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/config/form.js]: change this back to `formConfig`

File: src/applications/disability-benefits/all-claims/pages/additionalRemarks781.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
+
+import { PtsdNameTitle } from '../content/ptsdClassification';
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/pages/additionalRemarks781.js]: use the new helper content `ptsd781NameTitle`

File: src/applications/disability-benefits/all-claims/pages/additionalRemarks781.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    </p>
+    <p>
+      If you have any supporting documents or buddy statements to support your
+      claim, you‘ll have a chance to upload those later in the application.
+    </p>
+  </div>
+);
+
+export const uiSchema = {
+  'ui:title': ({ formData }) => (
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/pages/additionalRemarks781.js]: If you use `ptsd781NameTitle` you can just do 
```'ui:title': pstd781NameTitle,```

File: src/applications/disability-benefits/all-claims/pages/additionalRemarks781.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
+import React from 'react';
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/pages/additionalRemarks781.js]: Can we make this file extension a `.jsx` since it's using react.

**Review by aaronjlech — CHANGES_REQUESTED** [review]
Just a few small tweaks.

File: src/applications/disability-benefits/all-claims/tests/config/schema.js [hunk 9]
```diff
@@ -0,0 +1,1099 @@
+// This file to be moved to vets-json-schema repo before form hits production
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/tests/config/schema.js]: Is there a reason this file is in `src/applications/disability-benefits/all-claims/tests/config/schema.js`

**Review by aaronjlech — CHANGES_REQUESTED** [review]

**Comment by zurbergram** [comment]
Closing this PR.  New PR for https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets-website/9158 is https://github.com/department-of-veterans-affairs/vets-website/pull/9206

**Pull request closed without merging**